test/metabase/test.clj

```diff
@@ -431,14 +431,23 @@
   "Impl for `with-fake-events-collector` macro."
   [f]
   (let [original-publish-event! events/publish-event!
-        collector               (atom [])]
+        collector               (atom [])
+        ;; when testing with mt/user-http-request we could receive an user-login event
+        ;; for the first time it's called for an user.
+        ;; since most of the event tests does not concern this, we should filter it out
+        ;; to avoid flaky tests
+        non-login-events        (fn [] (remove #(#{:user-login} (first %)) @collector))]
     (with-redefs [events/publish-event! (fn [topic event-item]
                                           (swap! collector conj [topic event-item])
                                           (original-publish-event! topic event-item))]
       (assert @#'events/events-initialized?, "events was not initialized")
-      (f {:events     (fn [] @collector)
-          :last-event #(last @collector)}))))
-
+      (f {:raw-events   (fn [] @collector) ;; provide a way to get raw all events
+          :events       non-login-events
+          :last-event   #(last (non-login-events))
+          :topic-count  (fn []
```

Comment:
`:topic-count` sounds like a single number. I would rename to `topic-counts` or `topic->count`

Reply:
I went with `topic->count` 88da822e5d